tests/api/test_remote_engine.py

```diff
@@ -58,7 +58,8 @@ def finished(self):
 
     monkeypatch.setattr(xcc.Job, "submit", mock_return(job))
     monkeypatch.setattr(xcc.Job, "result", result)
-    monkeypatch.setattr(xcc.Job, "_details", _details)
+    monkeypatch.setattr(xcc.Job, "clear", lambda _: None)
+    job._details = {"status": "open"}
```

Comment:
Very minor point, but I think it would look better if we assigned `job._details` immediately after `job` is instantiated:
```python
    job = xcc.Job(id_="123", connection=connection)
    job._details = {"status": "open"}l

    result = {"samples": np.array([[1, 2], [3, 4]]), "foo": [np.array([5, 6])]}

    monkeypatch.setattr(xcc.Job, "submit", mock_return(job))
    monkeypatch.setattr(xcc.Job, "result", result)
    monkeypatch.setattr(xcc.Job, "clear", mock_return(None))
    monkeypatch.setattr(xcc.Job, "finished", finished)
    return job
```

Reply:
True, I agree!